mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -266,6 +266,32 @@ LogicalResult tosa::AvgPool2dOp::verify() {
   return emitOpError("input/output element types are incompatible.");
 }
 
+LogicalResult tosa::ClampOp::verify() {
+  mlir::Type inputDT =
+      llvm::cast<ShapedType>(getInput().getType()).getElementType();
+  mlir::Type maxFpDT = getMaxFpAttr().getType();
+  mlir::Type minFpDT = getMinFpAttr().getType();
+  mlir::Type outputDT =
+      llvm::cast<ShapedType>(getOutput().getType()).getElementType();
+  unsigned dataTypeBitWidth = inputDT.getIntOrFloatBitWidth();
+
+  if (!(inputDT.getTypeID() == outputDT.getTypeID()))
```

Comment:
Can't we just compare the IO element types?

Reply:
Fixed